test/api/inputmap.test.js

```diff
@@ -0,0 +1,30 @@
+import { Generator } from '@jspm/generator';
+import assert from 'assert';
+
+const generator = new Generator({
+  mapUrl: import.meta.url,
+  inputMap: {
+    "imports": {
+      "react": "https://cdn.skypack.dev/react"
+    }
+  },
+  env: ['production', 'browser']
+});
+
+await generator.install('react-dom@17')
+
+const json = generator.getMap();
+
+assert.deepEqual(json, {
+  "imports": {
+    "react": "https://cdn.skypack.dev/react",
+    "react-dom": "https://ga.jspm.io/npm:react-dom@17.0.2/index.js"
+  },
+  "scopes": {
+    "https://ga.jspm.io/": {
+      "object-assign": "https://ga.jspm.io/npm:object-assign@4.1.1/index.js",
+      "react": "https://ga.jspm.io/npm:react@17.0.2/index.js",
```

Comment:
I guess we should probably dedupe this? Let's post a bug to track.

Reply:
Posted https://github.com/jspm/generator/issues/36.

Reply:
yeah it's a good question what should be the default behaviour. We don't really know which version the "skypack" `react` is, is it safe to try and dedupe it or not